sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -82,54 +86,58 @@ int64_t MemoryBodyStream::OnRead(uint8_t* buffer, int64_t count, Context const&
   return copy_length;
 }
 
-#if defined(AZ_PLATFORM_POSIX)
-int64_t FileBodyStream::OnRead(uint8_t* buffer, int64_t count, Azure::Core::Context const& context)
+#if defined(AZ_PLATFORM_WINDOWS)
+#pragma warning(push)
+// warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead.
+#pragma warning(disable : 4996)
+#endif
+
+FileBodyStream::FileBodyStream(const std::string& filename) : m_offset(0)
 {
-  (void)context;
-  auto result = pread(
-      this->m_fd,
-      buffer,
-      std::min(count, this->m_length - this->m_offset),
-      this->m_baseOffset + this->m_offset);
+  m_fileStreamHolder = fopen(filename.c_str(), "rb");
```

Comment:
open/CreateFile. For CreateFile, we should discuss flags - they're very important